lib/config/error-codes/index.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Kuzzle, a backend software, self-hostable and ready to use
+ * to power modern apps
+ *
+ * Copyright 2015-2018 Kuzzle
+ * mailto: support AT kuzzle.io
+ * website: http://kuzzle.io
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+const { InternalError: KuzzleInternalError } = require('kuzzle-common-objects').errors;
+const api = require('./api');
+
+/** Check the format of the error-codes config files
+ *  @param {object} - error config file domain
+ */
+
+function checkErrorCodes(errorCodes) {
+
+  for (const domain of Object.keys(errorCodes)) {
+    const domainName = errorCodes[domain];
+    if (!(domainName.hasOwnProperty('subdomains')) || typeof domainName['subdomains'] !== 'object' 
+      || !(domainName.hasOwnProperty('code')) || typeof domainName['code'] !== 'number' 
+      || !(domainName['code'] > 0 && domainName['code'] < 10)) {
+      throw new KuzzleInternalError(`Field ${domain} wrongly formatted.`, 'Invalid error file', 1);
```

Comment:
```suggestion
      throw new KuzzleInternalError(`Format error on error domain "${domain}".`, 'Invalid error file', 1);
```

Personnally, I would prefer simpler "ifs" condition, each one raising an intelligible error (code out of range, missing subdomain, etc), rather than one big "catch-all if" raising a global error, because this can be hard to figure what the error really is for plugin developers (and for us too)
For instance, if someone puts a "1027" code, it can be very hard to understand that the error comes from that.

Reply:
I see what you mean by being the more specific you can, but imho it's not hard to figure where is the error since the object is pointed
Adding a "ifs" forest would give a messy code don't you think ?

Reply:
I personnally prefer a more precise erroring system: in my experience this leads to easier to understand messages, ergo customers ask less questions and we have less support to do.

But I don't feel like battling over it either.

Reply:
Okay, will do